commands/build.go

```diff
@@ -649,3 +657,154 @@ func dockerUlimitToControllerUlimit(u *dockeropts.UlimitOpt) *controllerapi.Ulim
 	}
 	return &controllerapi.UlimitOpt{Values: values}
 }
+
+// resolvePaths resolves all paths contained in controllerapi.BuildOptions
+// and replaces them to absolute paths.
+func resolvePaths(options *controllerapi.BuildOptions) (_ *controllerapi.BuildOptions, err error) {
+	if options.ContextPath != "-" {
+		options.ContextPath, err = absIfLocal(options.ContextPath)
```

Comment:
I'm not sure if we should do `os.Stat` here as part of `absIfLocal`? I think we should just do `filepath.Abs`, and similarly below.

We know at this point it's definitely a path, and if the file doesn't exist, we should let the controller throw the error (which may or may not be a file not found error, we shouldn't assume the implementation at this point).

Reply:
Thank you for the comment. Yes, we don't need os.Stat here because the options are already strongly typed. Fixed the patch.

Reply:
I just pushed a commit on top to handle the new oci-layout named context (which can also be a path). Does that seem good to you? If so, I think this is ready-to-go :tada:

Reply:
Thank you for the following-up! LGTM.